common/dynamicconfig/collection.go

```diff
@@ -414,6 +420,84 @@ func (c *Collection) GetBoolPropertyFilteredByTaskQueueInfo(key Key, defaultValu
 	}
 }
 
+// GetBoolPropertyFilteredByDestination gets property with destination as filter and asserts that it's a bool
+func (c *Collection) GetBoolPropertyFilteredByDestination(key Key, defaultValue any) BoolPropertyFnWithDestionationFilter {
```

Comment:
in my revision, default values are on the setting and not the call, and they are typed.

it's `any` here because you can pass a `[]ConstrainedValue` instead of a single value to have complex defaults. the new api has a different (also type-safe) to do that.